src/lxc/conf.c

```diff
@@ -3936,29 +3940,60 @@ static void turn_into_dependent_mounts(const struct lxc_rootfs *rootfs)
 	 */
 	move_fd(memfd);
 
+	char p_mount[PATH_MAX], p_path[PATH_MAX];
+	bool p_mount_shared, p_path_shared;
+
+	p_mount[0] = '\0';
+	p_path[0] = '\0';
+
 	while (getline(&line, &len, f) != -1) {
 		char *opts, *target;
+		bool shared;
+
+		TRACE("mountinfo: %s", line);
 
 		target = get_field(line, 4);
 		if (!target)
 			continue;
 
 		opts = get_field(target, 2);
-		if (!opts)
-			continue;
-
 		null_endofword(opts);
-		if (!strstr(opts, "shared"))
-			continue;
+
+		if (opts && strstr(opts, "shared"))
+			shared = true;
 
 		null_endofword(target);
-		ret = mount(NULL, target, NULL, MS_SLAVE, NULL);
+
+		// Find parent mount of rootfs->path (longest prefix match)
```

Comment:
you don't need to do this string comparisons to find parent mount, because `/proc/self/mountinfo` file provides you with this data ;-)

For example:
```
root@ubuntu:/# cat /proc/self/mountinfo
3541 156 259:4 /lxc/ubuntu/rootfs / rw,relatime master:120 - ext4 /dev/nvme1n1p4 rw
3896 3541 0:138 / /dev rw,relatime - tmpfs none rw,size=492k,mode=755,uid=100000,gid=100000,inode64
3897 3541 0:153 / /proc rw,nosuid,nodev,noexec,relatime - proc proc rw
3963 3964 0:153 /sys/net /proc/sys/net rw,nosuid,nodev,noexec,relatime - proc proc rw
```

A first column is the unique `mnt_id` of a mount (host-wide!). A second column shows parent `mnt_id` of the mount.
`/` (`mnt_id = 3541`) is obviously the parent mount for `/dev` (`mnt_id = 3896`) and `/proc` (`mnt_id = 3897`). As you can see both `/dev` and `/proc` have `3541` in the 2nd column (parent mount `mnt_id`).

And also using path-based way of finding parent mounts is not reliable because somebody can create an overmount, for example.